src/webgpu/api/validation/capability_checks/limits/maxStorageBufferBindingSize.spec.ts

```diff
@@ -0,0 +1,166 @@
+  }
+}
+
+function getTestValue(testValueName: TestValue, requestedLimit: number) {
+  switch (testValueName) {
+    case 'atLimit':
+      return roundDown(requestedLimit, kStorageBufferRequiredSizeAlignment);
+    case 'overLimit':
+      // Note: the requestedLimit might not meet alignment requirements.
+      return roundDown(
+        requestedLimit + kStorageBufferRequiredSizeAlignment,
+        kStorageBufferRequiredSizeAlignment
+      );
+  }
+}
+
+function getDeviceLimitToRequestAndValueToTest(
+  limitValueTest: LimitValueTest,
+  testValueName: TestValue,
+  defaultLimit: number,
+  maximumLimit: number
+) {
+  const requestedLimit = getDeviceLimitToRequest(limitValueTest, defaultLimit, maximumLimit);
+  return {
+    requestedLimit,
+    testValue: getTestValue(testValueName, requestedLimit),
+  };
+}
+
+const limit = 'maxStorageBufferBindingSize';
+export const { g, description } = makeLimitTestGroup(limit);
+
+g.test('createBindGroup,at_over')
+  .desc(`Test using createBindGroup at and over ${limit} limit`)
+  .params(kLimitBaseParams.combine('bufferPart', kBufferPartsKeys))
+  .fn(async t => {
+    const { limitTest, testValueName, bufferPart } = t.params;
+    const { adapter, defaultLimit, maximumLimit } = await t.getAdapterAndLimits();
+    const { requestedLimit, testValue } = getDeviceLimitToRequestAndValueToTest(
+      limitTest,
+      testValueName,
+      defaultLimit,
+      maximumLimit
+    );
+
+    await t.testDeviceWithSpecificLimits(
+      adapter,
+      requestedLimit,
+      testValue,
+      async ({ device, testValue, shouldError }) => {
+        const bindGroupLayout = device.createBindGroupLayout({
+          entries: [
+            {
+              binding: 0,
+              visibility: GPUShaderStage.COMPUTE,
+              buffer: { type: 'storage' },
+            },
+          ],
+        });
+
+        const { size, offset } = getSizeAndOffsetForBufferPart(device, bufferPart, testValue);
+        device.pushErrorScope('out-of-memory');
+        const storageBuffer = device.createBuffer({
+          usage: GPUBufferUsage.STORAGE,
+          size,
+        });
+        const outOfMemoryError = await device.popErrorScope();
+
+        if (!outOfMemoryError) {
+          await t.expectValidationError(
+            () => {
+              device.createBindGroup({
+                layout: bindGroupLayout,
+                entries: [
+                  {
+                    binding: 0,
+                    resource: {
+                      buffer: storageBuffer,
+                      ...(offset && { offset }),
+                      size: testValue,
+                    },
+                  },
+                ],
+              });
+            },
+            shouldError,
+            `size: ${size}, offset: ${offset}, testValue: ${testValue}`
+          );
+
+          storageBuffer.destroy();
+        }
+      }
+    );
+  });
+
+g.test('validate,maxBufferSize')
```

Comment:
Hmm, personally I think that 'validate' is redundant here because this is known to be a validation test already. Initially I thought maybe 'greaterThanOrEqualToMaxBufferSize'? But that seems really long... I would want to adopt something like the matcher names in gtest, i.e. lt -> less than, le -> less than or equal and then we could do: 'le,maxBufferSize' and 'ge,32'. But idk if we can make that a standard. (Ditto for the other file as well)

Reply:
I see your point. @kainino0x had me name it that in another PR so I've been following that pattern.